libs/martech/martech.js

```diff
@@ -97,49 +98,89 @@ const getDtmLib = (env) => ({
       : env.consumer?.marTechUrl || 'https://assets.adobedtm.com/d4d114c60e50/a0e989131fd5/launch-a27b33fc2dc0-development.min.js',
 });
 
-export default async function init({ persEnabled = false, persManifests }) {
+const setupEntitlementCallback = () => {
+  const setEntitlements = async (destinations) => {
+    const { default: parseEntitlements } = await import('../features/personalization/entitlements.js');
+    return parseEntitlements(destinations);
+  };
+
+  const getEntitlements = () => new Promise((resolve) => {
```

Comment:
I think there are use-cases where the Promise would fail, we need to handle those as well. I checked https://cpeyer-aep--milo--adobecom.hlx.page/ in Incognito and I'm stuck with a pending promise. We need to set a timeout or fail directly if `alloy` object is not available or other Martech logic is not running because of extension or hosts blockers

Reply:
Very good point.  I've added a timeout to it where it will just set it to an empty array in that case.  I think that's better than a rejection, but would like your opinion on it.  Timeout is currently set to 3s.

Reply:
A default value should be enough, we've used the same approach in FEDS code. Thanks!